src/google/protobuf/compiler/objectivec/objectivec_enum.cc

```diff
@@ -44,13 +45,32 @@ namespace objectivec {
 EnumGenerator::EnumGenerator(const EnumDescriptor* descriptor)
     : descriptor_(descriptor),
       name_(EnumName(descriptor_)) {
+  // Track the names for the enum values, and if an alias overlaps a base
+  // value, skip making a name for it. Likewise if two alias overlap, the
+  // first one wins.
+  // The one gap in this logic is if two base values overlap, but for that
+  // to happen you have to have "Foo" and "FOO" or "FOO_BAR" and "FOO_BAR",
```

Comment:
I don't see the difference between FOO_BAR and FOO_BAR.

Reply:
fixed